table/queries/merged_reduced_scans.sql

```diff
@@ -21,6 +21,152 @@ CREATE TEMP FUNCTION CleanError(error STRING) AS (
+  IF(error = "Get http://[IP]: dial tcp [IP]:[PORT]: getsockopt: connection reset by peer", "SUCCESS",
+  IF(error = "Get http://[IP]: dial tcp [IP]:[PORT]: getsockopt: network is unreachable", "NET_UNREACHABLE",
+  IF(error = "Get http://[IP]: dial tcp [IP]:[PORT]: getsockopt: no route to host", "NO_ROUTE_TO_HOST",
+  IF(error = "Get http://[IP]: dial tcp [IP]:[PORT]: socket: too many open files", "INVALID",
+  IF(error = "Get http://[IP]: EOF", "SUCCESS",
+  IF(error = "Get http://[IP]: net/http: request canceled (Client.Timeout exceeded while awaiting headers)", "SUCCESS",
+  IF(error = "Get http://[IP]: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)", "TIMEOUT",
+  IF(error = "Get http://[IP]: read tcp [IP]:[PORT]->[IP]:[PORT]: read: connection reset by peer", "SUCCESS",
+  IF(error = "Get http://[IP]: unexpected EOF", "SUCCESS",
+  IF(error = "i/o timeout", "SUCCESS",
+  IF(error = "Incorrect echo response", "SUCCESS",
+  IF(error = "Incorrect web response: bodies don't match", "SUCCESS",
+  IF(error = "Incorrect web response: certificates don't match", "SUCCESS",
+  IF(error = "Incorrect web response: cipher suites don't match", "SUCCESS",
+  IF(error = "Incorrect web response: status lines don't match", "SUCCESS",
+  IF(error = "network is unreachable", "NET_UNREACHABLE",
+  IF(error = "no route to host", "NO_ROUTE_TO_HOST",
+  IF(error = "null", "SUCCESS",
+  IF(error = "protocol not available", "INVALID",
+  IF(error = "read tcp [IP]:[PORT]->[IP]:[PORT]: i/o timeout", "SUCCESS",
+  IF(error = "read tcp [IP]:[PORT]->[IP]:[PORT]: read: connection reset by peer", "SUCCESS",
+  IF(error = "Received response", "SUCCESS",
+  IF(error = "too many open files", "INVALID",
+  IF(error = "write tcp [IP]:[PORT]->[IP]:[PORT]: write: broken pipe", "SUCCESS",
+  IF(error = "write tcp [IP]:[PORT]->[IP]:[PORT]: write: connection reset by peer", "SUCCESS",
+  IF(error = "write tcp [IP]:[PORT]: write: connection reset by peer", "SUCCESS",
+  IF(REGEXP_CONTAINS(error, "response missing Location header"), "SUCCESS",
+  IF(REGEXP_CONTAINS(error, "bad Content-Length "), "SUCCESS",
+  IF(REGEXP_CONTAINS(error, "failed to parse Location header"), "SUCCESS",
+  IF(REGEXP_CONTAINS(error, "malformed HTTP response"), "SUCCESS",
+  IF(REGEXP_CONTAINS(error, "malformed HTTP status code"), "SUCCESS",
+  IF(REGEXP_CONTAINS(error, "malformed HTTP version"), "SUCCESS",
+  IF(REGEXP_CONTAINS(error, "malformed MIME header line"), "SUCCESS",
+  IF(REGEXP_CONTAINS(error, "tls: received record with version"), "SUCCESS",
+     "UNKNOWN")))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))
+);
+
+
+# Classify errors we believe occured after the domain was sent in the probe.
+# Input is a string returned from CleanError
+# Output is a small set of classification strings.
+CREATE TEMP FUNCTION ClassifyProbeError(error STRING) AS (
```

Comment:
- Call ClassifyDialError and return INVALID if the output is not SUCCESS, to avoid duplication.
- Use the CASE expression
- The different probe types behave differently. We need a function for each. Reusing the Dial one should help.

Reply:
switched to case/when,
used ClassifyDialError to find invalid cases

I didn't find specific cases which had different probe errors for the same response string, but I did find a lot of overlap. (https probes can also get all the http errors for example) Do you have a specific example of a case where the probe error should be different between scan types?

Reply:
Receiving a response have a different meaning in Discard, versus the others.
EOF is expected in Discard, but not the others.
HTTP response mismatches are not really failures in HTTPS, since you can't manipulate the HTTPS response.

Reply:
There are things we can do to avoid the duplication. the ClassifyDialError is an example. Perhaps we can have something for write error, read error, http error, https error, validation error. This requires some further modeling of the probe behavior, but I think this seems reasonable:
1. Dial
2. Write request
3. Read response
5. Validate response